SubterfugeCore/Core/Network/NetworkResponse.cs

```diff
@@ -1,19 +1,91 @@
-﻿namespace SubterfugeCore.Core.Network
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace SubterfugeCore.Core.Network
 {
     /// <summary>
     /// Base Network response. All Network responses should have a 'success' variable to determine if the operation
     /// succeeded and a message describing any issues if it didn't
     /// </summary>
-    public class NetworkResponse
+    public class NetworkResponse<T>
     {
         /// <summary>
-        /// If the network operation was successful
+        /// The HttpResponse object that comes directly from the HttpClient
         /// </summary>
-        public bool Success { get; set; }
+        public HttpResponseMessage HttpResponse { get; set; }
         
         /// <summary>
-        /// Details about the network operation
+        /// If the request resulted in an error, this variable gets populated with the error content.
         /// </summary>
-        public string Message { get; set; }
+        public NetworkError ErrorContent { get; set; }
+        
+        /// <summary>
+        /// The raw string of the HttpResponse. This may be an error string or a valid JSON string.
+        /// </summary>
+        public string ResponseContent { get; set; }
+        
+        /// <summary>
+        /// A templated object that holds the parsed JSON response if the request was successful.
+        /// </summary>
+        public T Response { get; set; }
+        
+        /// <summary>
+        /// Static method to create a new Network Response.
+        /// </summary>
+        /// <param name="responseMessage">The HttpResponseMessage that is recieved from the HttpClient</param>
+        /// <returns>A new instance of a NetworkResponse</returns>
+        public static async Task<NetworkResponse<T>> FromHttpResponse(HttpResponseMessage responseMessage)
+        {
+            NetworkResponse<T> response = new NetworkResponse<T>();
+            response.HttpResponse = responseMessage;
+            
+            // Read the response
+            string responseContent = await responseMessage.Content.ReadAsStringAsync();
+            response.ResponseContent = responseContent;
+
+            // Deserialize the response if it was successful.
+            if (responseMessage.IsSuccessStatusCode)
+            {
+                T responseTemplate = default(T);
+                try
+                {
+                    responseTemplate = JsonConvert.DeserializeObject<T>(responseContent);
+                }
+                catch (JsonException e)
+                {
```

Comment:
I don't like an empty catch, I believe it is an anti-pattern. If the response wasn't a valid JSON, we should report that somehow (even if it's just a logging statement or something of the sort) so that it will be easy to check what the root of the error was.